lesson16/main.js

```diff
@@ -188,14 +230,31 @@ function withinThreeDays(day) {
   return diff < msInThreeDays;
 }
 
-function addClickEventListener() {
-  tabArea.addEventListener("click", (e) => {
-    const targetElement = e.target;
-    if (targetElement === e.currentTarget) return;
-    tabArea.querySelector(".active").classList.remove("active");
-    targetElement.classList.add("active");
-    articleArea.querySelector(".active").classList.remove("active");
-    document.getElementById(targetElement.dataset.id).classList.add("active");
+function addClickEventChangeElement(
+  parentElem,
+  isChangeClickedElem,
+  parentOfRelationElem
+) {
+  parentElem.addEventListener("click", (e) => {

```

Comment:
Q. Isn't it not always the parent element?
```suggestion
  parentElem.addEventListener("click", (e) => {

```

Reply:
読みづらいコードで申し訳ありません。
タブの切替とコメントの切替に使用しています。

function addClickEventChangeElement(
  parentElem,　　　　　　　　　　　　　
  **//タブまたはコメントアイコンのラッパー（親div）です。**
  isChangeClickedElem,
  **//タブ切替に適用する場合trueとしています。**　　　　　
  parentOfRelationElem　　
 **//記事またはコメントのラッパー（親div）です。**
) {
  parentElem.addEventListener("click", (e) => {
    if (e.target === e.currentTarget) return;
   **//ラッパーをクリックしていたら中止。**
    if (isChangeClickedElem) {
   **// タブ切替時のみ実行しています。**   parentElem.querySelector(".active").classList.remove("active");
      e.target.classList.add("active");
 **//タブのactiveクラスを付け替えています。**
    }
    if (parentOfRelationElem.querySelector(".active")) {
      parentOfRelationElem.querySelector(".active").classList.remove("active");
　**//記事またはコメントのactiveクラスを除去しています。**
    }
    if (document.getElementById(e.target.dataset.id)) {
      document.getElementById(e.target.dataset.id).classList.add("active");
**//関連する記事またはコメントにactiveクラスを付与しています。**
    }
  });
}

Reply:
IMO
I thought it was important to make it a function, but I also think it is good to define it separately.
私の考えです。IMOなので修正するかはお任せします！
```js
const addEventCommentIcon = (commentIcon) => {
  commentIcon.addEventListener("click", (event) => {
    const activeCommentContainer = document.querySelector(
      ".comment-container.active"
    );
    if (activeCommentContainer)
      activeCommentContainer.classList.remove("active");
    document.getElementById(event.target.dataset.id).classList.add("active");
  });
};
```

```js
function addEventTabArea(tabArea) {
  tabArea.addEventListener("click", (e) => {
    if (e.target === e.currentTarget) return;

    tabArea.querySelector(".active").classList.remove("active");
    e.target.classList.add("active");

    const activeTabContent = document.querySelector(".genre-container.active");
    if (activeTabContent) {
      activeTabContent.classList.remove("active");
      document.getElementById(e.target.dataset.id).classList.add("active");
    }
  });
}

```

Reply:
ここまでしていただいて申し訳ありませんが、現在のもので進めさせていただきたいと思います。
少し修正を加えさせていただきました。
874895f8ff89d3c74afef536a18ed94adc3a65e1